### DISCUSSION TITLE
Attributes on ProductOption & ProductOptionValue

### DISCUSSION THREAD

**Discussion by charlielangridge** [lunarphp/lunar#718] [Feature Requests]
I'd really like the functionailty to add attributes to ProductOption and ProductOptionValue. 
Use case example of these would be: 
ProductOption - being able to define what type of option it was so it can be used to dictate what type of selection input it is on the front end (radio box, colour picker, dropdown etc)
ProductionOptionValue - being able to add details to an option - for example if the option was for a paper size, the name would be "A4", but the detail might have "297x210mm", or sorting order details

I can work on a PR for this if the feature is wanted?

**Comment by glennjacobs** [comment]
@charlielangridge I think that would be very handy. 

I'm just thinking about how we use this for the storefront, so the UI knows to display options in a certain way. E.g. we could have `T-Shirt Colours` and `Trouser Colours`, but they'd want to display the same on the frontend.

Yes, you could add a "option type" attribute, but it would need to be set on all options.

I feel like we should consider creating "Option Types", along the lines of "Product Types". This way we know the "type" for display purposes and when creating a new option it would only show the relevant attributes.

**Comment by glennjacobs** [comment]
Also worth noting we have this outstanding PR which I guess we'd look to build upon https://github.com/lunarphp/lunar/pull/419

**Reply by charlielangridge** [reply]
@glennjacobs - is that due to be merged into 0.2? or should i fork off that fork? I'm keen not to step on anyone elses toes - or - recreating work that's been done and just isnt in the main branch yet

**Reply by glennjacobs** [reply]
I think it's best we try to get that merged into `main` so then you can confidently develop on it. @alecritson could you take a look at that existing PR?

I'll convert this discussion to an issue also.
